README.md

```diff
@@ -2,31 +2,58 @@
 
 Implementation of a [TileSource](https://openseadragon.github.io/docs/OpenSeadragon.TileSource.html) for [OpenSeadragon](https://openseadragon.github.io/) based on [geotiff.js](https://geotiffjs.github.io/), enabling local and remote TIFF files to be viewed without using an image server.
 
+Implements an ES6 module extending the original work from [pearcetm/GeoTIFFTileSource](https://github.com/pearcetm/GeoTIFFTileSource). For further information on creating GeoTIFF files and it's requirements, please see the original repository.
 
 
-See it in action at [https://pearcetm.github.io/GeoTIFFTileSource/demo/demo.html](https://pearcetm.github.io/GeoTIFFTileSource/demo/demo.html)
+## Usage
 
-## How to create GeoTIFF files
+### Installation
 
-In order to generate a GeoTIFF file compatible with this library, you can for instance use [sharp](https://sharp.pixelplumbing.com/), a *High performance Node.js image processing* library.
+The plugin is available as an npm package:
+
+```bash
+npm i geotiff-tilesource
+```
+The plugin requires OpenSeadragon 3.0.0 or above, and includes [geotiff.js](https://geotiffjs.github.io/) as a dependency.
+
+### Extending OpenSeadragon
+
+The plugin allows for extending OpenSeadragon with a new `TileSource` type, the `GeoTIFFTileSource`. To initialize, call the following at the top-most import of OpenSeadragon:
 
 ```javascript
-import sharp from 'sharp'
+import OpenSeadragon from 'openseadragon';
+import { enableGeoTIFFTileSource } from "geotiff-tilesource";
 
-sharp('input.jpg', {limitInputPixels:false})
-    .tiff({tile:true, pyramid:true})
-    .toFile('output.tiff')
-    .then(console.log)
-    .catch(console.error);
+enableGeoTIFFTileSource(OpenSeadragon);
 ```
 
-Many options are available. 
+This will make the `OpenSeadragon.GeoTIFFTileSource` class available for use.
+
+### Using the GeoTIFFTileSource
 
-[Check sharp documentation on TIFF output format.](https://sharp.pixelplumbing.com/api-output#tiff)
+#### Prepare TileSources
 
+GeoTIFFTileSource accepts both local and remote GeoTIFF files. For local files, the `url` parameter should be a `File` object. For remote files, the `url` parameter should be a string. The `getAllTileSources` reads a local or remote GeoTIFF file and returns an array of `OpenSeadragon.GeoTIFFTileSource` objects, one for each image (page) in the GeoTIFF file.
 
-## Prerequisites for remote files - HTTP range requests
```

Comment:
I think this is still useful information. It explains to developers why not all remote TIFF files will work.